moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -203,7 +203,7 @@ bool planning_pipeline::PlanningPipeline::generatePlan(const planning_scene::Pla
   // broadcast the request we are about to work on, if needed
   if (publish_received_requests)
   {
-    received_request_publisher_->publish(req);
+    received_request_publisher_->publish(req.toMessage());
```

Comment:
Related to my other comment about why `toMessage` is needed, since the request struct inherits from the message.
Maybe you need to reinterpret_cast to the base class?